pandas/io/pickle.py

```diff
@@ -143,7 +143,8 @@ def read_pickle(
 
     Returns
     -------
-    same type as object stored in file
+    object
+        Same type as object stored in file.
```

Comment:
Do you mind checking if `read_pickle` can really load any type (object), or if we restrict it to return a `DataFrame`, `Series`...? It would be good to be more specific with the return type if makes sense.

Also, can you improve the existing description to say what it's being returned, not what's the type please? Same as in my other comment.

Reply:
pandas.read_pickle takes a "String, path object (implementing os.PathLike[str]), or file-like object implementing a binary readlines() function. Also accepts URL. URL is not limited to S3 and GCS." and returns the "same type as object stored in file"

We are not placing any restriction on it (i.e. DataFrame, Series, etc).

I'll follow the suggested improvement from groupby feedback above.

Reply:
If I understand correctly, the text you quote refers to the input of the function, what we need to document is the output. I checked, and seems `read_pickle` will allow to load any Python object. For example, if I have a file with an int, `read_pickle` will just load it:

```
>>> import pandas
>>> pandas.read_pickle('int.pickle')
10
```

Feels a bit strange, but I guess it makes sense. You can leave the `object` type in the return type. Maybe in the description you can mention that the pickle object can be a type that is not a pandas type.

Reply:
> If I understand correctly, the text you quote refers to the input of the function, 

Yes, this is correct. 

To document is the output more clearly, does these seem like appropriate changes?

Description: 
```
    Load pickled pandas object (or any object) from file and return the unpickled object.
```
Return: 
```
    Returns
    -------
    object
        The unpickled pandas (or any object) that was stored in file.
```

Reply:
Looks good, thank you!